File: tournaments/matrixTransposition/matrixTransposition.js
function matrixTransposition(matrix) {
    const rows = matrix.length;
    const cols = matrix[0].length;
    const result = new Array(cols);

    for (let col = 0; cols > col; ++col) {
        result[col] = new Array(rows);

        for (let row = 0; rows > row; ++row) {
            result[col][row] = matrix[row][col];
        }
    }

    return result;
}

File: tournaments/matrixTransposition/matrixTransposition.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub matrixTransposition {
    my ($matrix) = @_;

    my $rows = @$matrix;
    my $cols = @{@$matrix[0]};
    my @result;

    for (my $col = 0; $cols > $col; ++$col) {
        push(@result, []);
        for (my $row = 0; $rows > $row; ++$row) {
            push(@{@result[$col]}, @{@$matrix[$row]}[$col]);
        }
    }
    return \@result;
}
